src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GetGoalTest.kt

```diff
@@ -0,0 +1,131 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.resource
+
+import org.junit.jupiter.api.Test
+import org.springframework.http.HttpStatus
+import uk.gov.justice.digital.hmpps.domain.aValidPrisonNumber
+import uk.gov.justice.digital.hmpps.domain.aValidReference
+import uk.gov.justice.digital.hmpps.domain.anotherValidPrisonNumber
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.aValidTokenWithNoAuthorities
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.aValidTokenWithViewAuthority
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.bearerToken
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.ErrorResponse
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.GoalResponse
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.actionplan.aValidCreateGoalRequest
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.actionplan.assertThat
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.assertThat
+
+class GetGoalTest : IntegrationTestBase() {
+  companion object {
+    private const val URI_TEMPLATE = "/action-plans/{prisonNumber}/goals/{goalReference}"
+    private val prisonNumber = aValidPrisonNumber()
+  }
+
+  @Test
+  fun `should return unauthorized given no bearer token`() {
+    val goalReference = aValidReference()
+
+    webTestClient.get()
+      .uri(URI_TEMPLATE, prisonNumber, goalReference)
+      .exchange()
+      .expectStatus()
+      .isUnauthorized
+  }
+
+  @Test
+  fun `should return forbidden given bearer token without required role`() {
+    // Given
+    val goalReference = aValidReference()
+
+    // When
+    val response = webTestClient.get()
+      .uri(URI_TEMPLATE, prisonNumber, goalReference)
+      .bearerToken(aValidTokenWithNoAuthorities(privateKey = keyPair.private))
+      .exchange()
+      .expectStatus()
+      .isForbidden
+      .returnResult(ErrorResponse::class.java)
+
+    // Then
+    val actual = response.responseBody.blockFirst()
+    assertThat(actual)
+      .hasStatus(HttpStatus.FORBIDDEN.value())
+      .hasUserMessage("Access Denied")
+      .hasDeveloperMessage("Access denied on uri=/action-plans/$prisonNumber/goals/$goalReference")
+  }
```

Comment:
Not for this PR but I wonder if there's scope to create a common set of tests for endpoints (no token, wrong role, malformed, etc.).